cmd/system-probe/modules/network_tracer.go

```diff
@@ -193,14 +194,17 @@ func (nt *networkTracer) Register(httpMux *module.Router) error {
 			maps = strings.Split(listMaps, ",")
 		}
 
-		ebpfMaps, err := nt.tracer.DebugEBPFMaps(maps...)
+		var ebpfMaps bytes.Buffer
+
+		err := nt.tracer.DebugEBPFMaps(&ebpfMaps, maps...)
+
 		if err != nil {
 			log.Errorf("unable to retrieve eBPF maps: %s", err)
 			w.WriteHeader(500)
 			return
 		}
 
-		utils.WriteAsJSON(w, ebpfMaps)
+		utils.WriteAsJSON(w, ebpfMaps.String())
```

Comment:
I don't think there is a strong need to keep this a JSON string (`test` gets output as `"test"`). So I think you should be able to pass `w` directly to `nt.tracer.DebugEBPFMaps` and get rid of the `bytes.Buffer` altogether.

Reply:
I haven't used these endpoints myself in a while, but I could see it being handy to keep json output so we can pipe into `jq` to get the stats we care about. 

Though this change won't actually preserve that; I'm guessing it will write the contents of `ebpfMaps`  as one giant escaped JSON string.

Reply:
Nothing below within the string is JSON formatted (it uses spew, see [example output](https://github.com/davecgh/go-spew?tab=readme-ov-file#sample-dump-output))

Reply:
As Bryce says it just outputs the string as a quoted text, I kept it with JSON though just in case anyone was depending on that. If you think it's not important to have those quotes I'll change it to use the writer directly.